app/src/main/java/com/olympics/olympicsandroid/model/presentationModel/helper/CountryEventsHelper.java

```diff
@@ -1,87 +1,200 @@
+     * The aim of this method is to create a CountryEventUnitModel = which will have the list of
+     * unique athletes participating for that country.
      *  and also events #DateSportsModel sorted by date.
      */
-    public CountryEventUnitModel getCountrySpecificData()
-    {
+    public CountryEventUnitModel createCountryEventUnitModel() {
 
         CountryEventUnitModel countryEventUnitModel = new CountryEventUnitModel();
 
         countryEventUnitModel.setCountryAlias(countryProfileEvents.getOrganization().getAlias());
-        countryEventUnitModel.setCountryName(countryProfileEvents.getOrganization().getDescription());
+        countryEventUnitModel.setCountryName(countryProfileEvents.getOrganization()
+                .getDescription());
         countryEventUnitModel.setCountryID(countryProfileEvents.getOrganization().getId());
 
+        countryEventUnitModel.setAthleteList(getAthleteListForCountry());
 
-        countryEventUnitModel.setDatesCountryMapping(getDatesofCountry());
+        // initialize mapping and then update with data
+        countryEventUnitModel.initializeEmptyDateSportsMapping();
+        updateDateSportsMapping(countryEventUnitModel.getDatesCountryMapping());
 
         return countryEventUnitModel;
     }
 
+    /**
+     * This method populates List of Athlete from CountryProfileEvents
+     * @return
+     */
+    private List<Athlete> getAthleteListForCountry() {
+
+        Set<Athlete> athleteList = new HashSet<>();
+        for (OlympicEvent everyEvent : countryProfileEvents.getOrganization().getEvents()) {
+            if (everyEvent == null || everyEvent.getParticipants() == null || everyEvent
+                    .getParticipants().isEmpty()) {
+                continue;
+            }
+
+            List<OlympicAthlete> participantList = everyEvent.getParticipants();
+            for (OlympicAthlete participant : participantList) {
+                if (participant == null) {
+                    continue;
+                }
+                Athlete athlete = new Athlete();
+                athlete.setAthleteName(participant.getPrint_name());
+                athlete.setAthleteGender(participant.getGender());
+                if (everyEvent.getSport() != null) {
+                    athlete.setSportName(everyEvent.getSport().getDescription());
+                }
+                athleteList.add(athlete);
+            }
+
+        }
+        return new ArrayList<>(athleteList);
+    }
+
+    /**
+     *
+     * @return map of <Unit_start_date,
+     */
+    private void updateDateSportsMapping(Map<String, DateSportsModel> dateSportsMapping) {
+
+        Map<String, OlympicEvent> allEventsMap = populateEventMapFromAllEventsSchedule();
+
+        // Get each event that selected country is participating in
+        for (OlympicEvent participatingEvent : countryProfileEvents.getOrganization().getEvents()) {
```

Comment:
We can add the athlete filtering here in this same loop, as the time complexity wont increase.